hnn_core/lfp.py

```diff
@@ -0,0 +1,234 @@
+"""
+LFPsim - Simulation scripts to compute Local Field Potentials (LFP) from cable
+compartmental models of neurons and networks implemented in NEURON simulation
+environment.
+
+LFPsim works reliably on biophysically detailed multi-compartmental neurons
+with ion channels in some or all compartments.
+
+Last updated 12-March-2016
+Developed by :
+Harilal Parasuram & Shyam Diwakar
+Computational Neuroscience & Neurophysiology Lab, School of Biotechnology,
+Amrita University, India.
+
+Email: harilalp@am.amrita.edu; shyam@amrita.edu
+www.amrita.edu/compneuro
+
+translated to Python and modified to use use_fast_imem by Sam Neymotin
+based on mhines code
+
+References
+----------
+1 . Parasuram H, Nair B, D'Angelo E, Hines M, Naldi G, Diwakar S (2016)
+Computational Modeling of Single Neuron Extracellular Electric Potentials
+and Network Local Field Potentials using LFPsim. Front Comput Neurosci 10:65
+[PubMed].
+"""
+
+# Authors: Mainak Jas <mjas@mgh.harvard.edu>
+#          Nick Tolley <nicholas_tolley@brown.edu>
+#          Sam Neymotin <samnemo@gmail.com>
+
+from neuron import h
+
+import numpy as np
+from numpy.linalg import norm
+
+
+def _get_sections_on_this_rank(sec_type='Pyr'):
+    ls = h.allsec()
+    ls = [s for s in ls if sec_type in s.name()]
+    return ls
+
+
+def _get_segment_counts(all_sections):
+    '''The segment count of a section excludes the endpoints (0, 1)'''
```

Comment:
```suggestion
    """The segment count of a section excludes the endpoints (0, 1)"""
```

I'm very nitpicky about code style, sorry!